terraform-aws-github-runner/modules/runners/lambdas/runners/src/scale-runners/scale-up-chron.ts

```diff
@@ -0,0 +1,109 @@
+import axios from 'axios';
+
+import { Config } from './config';
+import { getRepo, shuffleArrayInPlace } from './utils';
+import { ScaleUpChronMetrics } from './metrics';
+import { getRunnerTypes } from './gh-runners';
+import { sqsSendMessages } from './sqs';
+import { ActionRequestMessage, scaleUp} from './scale-up';
+import { randomUUID } from 'crypto';
+
+export async function scaleUpChron(metrics: ScaleUpChronMetrics): Promise<void> {
+  // This function does the following:
+  // 1. Queries for queued runners via HUD
+  // 2. Polls scale-config to filter the list to ones that are self-hosted by this fleet and
+  //    are ephemeral
+  // 3. Sends a SQS request to the scale-up lambda to provision more of those instances
+
+  let queuedJobs = await getQueuedJobs(metrics);
+
+  const scaleConfigRepo = getRepo(Config.Instance.scaleConfigOrg, Config.Instance.scaleConfigRepo);
+
+
+  const validRunnerTypes = await getRunnerTypes(scaleConfigRepo, metrics);
+
+  const minAutoScaleupDelayMinutes = 30;
+  // Only proactively scale up the jobs that have been queued for longer than normal
+  queuedJobs = queuedJobs.filter((runner) => {
+    return runner.min_queue_time_minutes >= minAutoScaleupDelayMinutes &&
+      runner.org === Config.Instance.scaleConfigOrg;
+  });
+
+  // Filter out the queued jobs that are do not correspond to a valid runner type
+  queuedJobs = queuedJobs.filter((requested_runner) => {
+    return Array.from(validRunnerTypes.keys()).some((available_runner_label) => {
+      return available_runner_label === requested_runner.runner_label;
+    });
+  });
+
+  // Send a message to the SQS queue to scale up the runners
+  let scaleUpRequests : Array<ActionRequestMessage> = queuedJobs.map((runner) => {
+    return {
+      "id": Math.floor(Math.random() * 100000000000000),
+      "eventType": "workflow_job",
+      "repositoryName": runner.repo,
+      "repositoryOwner": runner.org,
+      "runnerLabels": [runner.runner_label],
+    };
+  });
+
+  if (!Config.Instance.scaleUpRecordQueueUrl) {
+    throw new Error('scaleUpRecordQueueUrl is not set. Cannot send scale up requests');
+  }
+
+  for (const request of shuffleArrayInPlace(scaleUpRequests)) {
+    try{
+      await scaleUp("aws:sqs", request, metrics);
+      metrics.scaleUpChronSuccess();
+
+    } catch (error) {
+      metrics.scaleUpChronFailure((error as Error).message);
+    }
+
```

Comment:
send metrics on lambda execution success (something like scaleUpCronSuccess, to maintain the standard)